website/content/en/docs/olm-integration/generating-a-csv.md

```diff
@@ -1,208 +1,250 @@
 ---
-title: Generating a ClusterServiceVersion
-linkTitle: Generating a ClusterServiceVersion
-weight: 3
+title: Generating Manifests and Metadata
+linkTitle: Generating Manifests and Metadata
+weight: 20
 ---
 
-This document describes how to manage the following lifecycle for your Operator using the SDK's [`operator-sdk generate csv`][generate-csv-cli] command:
+This document describes how to manage packaging and shipping your Operator in the following stages:
+
+* **Generate your first release** - encapsulate the metadata needed to install your Operator with the
+[Operator Lifecycle Manager][olm] and configure the permissions it needs from the generated SDK files.
+* **Update your Operator** - apply any updates to Operator manifests made during development.
+* **Upgrade your Operator** - carry over any customizations you have made and ensure a rolling update to the
+next version of your Operator.
+
+## Overview
+
+Several `operator-sdk` subcommands manage operator-framework manifests, in particular [`ClusterServiceVersion`'s (CSVs)][doc-csv],
+for an Operator: [`generate bundle`][cli-gen-bundle], [`generate packagemanifests`][cli-gen-packagemanifests],
+and [`generate kustomize manifests`][cli-gen-kustomize-manifests]. See this [CLI overview][cli-overview] for
+details on each command.
+
+SDK projects are scaffolded with a `Makefile` containing the `bundle` recipe by default,
+which wraps `generate kustomize manifests`, `generate bundle`, and other related commands.
+Ideally the bundle format should be used as this is the default packaging format in operator-framework.
+However the package manifests format is still supported by operator-framework tooling.
+If your Operator is already formatted as a package manifests and you do not wish to migrate yet,
+you should add the following to your `Makefile` to make development easier:
+
+```make
+# Options for "packagemanifests".
+ifneq ($(origin CHANNEL), undefined)
+PKG_CHANNELS := --channel=$(CHANNEL)
+endif
+ifeq ($(IS_CHANNEL_DEFAULT), 1)
+PKG_IS_DEFAULT_CHANNEL := --default-channel
+endif
+PKG_MAN_OPTS ?= $(PKG_CHANNELS) $(PKG_IS_DEFAULT_CHANNEL)
+
+# Generate package manifests.
+packagemanifests: manifests
+  operator-sdk generate kustomize manifests -q
+  kustomize build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
+```
```

Comment:
Nit:
This bit on `Oh btw here's what to do ...` doesn't seem like it should be in the `Overview` section.
If we can't move it out of the way towards the end, can we put mark this more clearly with its own section?